File: app/src/main/java/com/simplemobiletools/musicplayer/receivers/ControlActionsListener.java
package com.simplemobiletools.musicplayer.receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.simplemobiletools.musicplayer.Constants;
import com.simplemobiletools.musicplayer.Utils;

public class ControlActionsListener extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        final String action = intent.getAction();
        switch (action) {
            case Constants.PREVIOUS:
            case Constants.PLAYPAUSE:
            case Constants.NEXT:
            case Constants.FINISH:
                Utils.sendIntent(context, action);
                break;
            default:
                break;
        }
    }
}

File: app/src/main/kotlin/com/simplemobiletools/musicplayer/receivers/ControlActionsListener.kt
package com.simplemobiletools.musicplayer.receivers

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.simplemobiletools.musicplayer.extensions.sendIntent
import com.simplemobiletools.musicplayer.helpers.*

class ControlActionsListener : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        when (val action = intent.action) {
            PREVIOUS, PLAYPAUSE, NEXT, FINISH, DISMISS -> context.sendIntent(action)
        }
    }
}
